### ISSUE TITLE
Enable URL auto-linking

### ISSUE THREAD

**Issue by jloeser** [hunyadi/md2conf#10]
See https://docs.gitlab.com/ee/user/markdown.html#url-auto-linking

This behavior corresponds to default Confluence behavior when you insert a simple URL.

Would it be nice to have this feature? Maybe this behavior can be enabled/disabled via cmdline option?

At the moment you need the following to have link in Confluence which is clickable:
```
See [https://foobar.com](https://foobar.com) for more information.
```
Instead of:
```
See https://foobar.com for more information.
```

On the other side there might be cases where you explicitly do not want a URL to be clickable :thinking:

**Comment by hunyadi** [comment]
This would likely require a regular expression to scan into text nodes in a Markdown AST. The regular expression should be robust against accidentally picking up something that may resemble but is not in fact a URL.

**Comment by jloeser** [comment]
Correct, but looks like this is not that easy. However, after quick search I found this here https://mathiasbynens.be/demo/url-regex where for example the following might be good enough:
```
@^(https?|ftp)://[^\s/$.?#].[^\s]*$@iS
```

**Comment by hunyadi** [comment]
The Markdown to Confluence converter is depending on the [Python-Markdown](https://github.com/Python-Markdown/markdown/) package for parsing Markdown documents into an HTML string, which is then parsed into an HTML element tree with *lxml*. I believe the best place to identify and capture "naked" links is in the Markdown parsing phase. I have found that the extension *MagicLink* in *PyMdown* adds [auto-linking](https://facelessuser.github.io/pymdown-extensions/extensions/magiclink/#auto-linking) support to *Python-Markdown*, which I have enabled. A naked link example has been added to the sample document.
